libraries/AC_PrecLand/AC_PrecLand.h

```diff
@@ -69,6 +74,32 @@ class AC_PrecLand
     // process a LANDING_TARGET mavlink message
     void handle_msg(const mavlink_landing_target_t &packet, uint32_t timestamp_ms);
 
+    // State of the Target Location
+    enum class State: uint8_t {
+        TARGET_NEVER_SEEN = 0,
+        TARGET_OUT_OF_RANGE,
+        TARGET_RECENTLY_LOST,
+        TARGET_FOUND
+    };
+
+    // return the last time PrecLand library had a output of the landing target position
+    uint32_t get_last_precland_output_ms() const { return _last_precland_output_ms; }
```

Comment:
How about calling this "get_last_valid_target_ms()"?

Maybe we can rename the "precland_output" part of this name.  The problem with "output" is that it depends upon your point of view.  One subsystem's "output" is another subsystems "input"... and everything in this library is "precland" so it's not adding value.